trade-finance-manager-ui/server/controllers/case/cancellation/cancel-cancellation.controller.ts

```diff
@@ -0,0 +1,42 @@
+import { Response } from 'express';
+import { CustomExpressRequest } from '@ukef/dtfs2-common';
+import { PRIMARY_NAVIGATION_KEYS } from '../../../constants';
+import { asUserSession } from '../../../helpers/express-session';
+import { CancelCancellationViewModel } from '../../../types/view-models';
+import api from '../../../api';
+import { canSubmissionTypeBeCancelled } from '../../helpers';
+
+export type CancelCancellationRequest = CustomExpressRequest<{ params: { _id: string } }>;
+
+/**
+ * controller to get the cancel cancellation page
+ *
+ * @param req - The express request
+ * @param res - The express response
+ */
+export const getCancelCancellation = async (req: CancelCancellationRequest, res: Response) => {
+  const { _id } = req.params;
+  const { user, userToken } = asUserSession(req.session);
+  try {
+    const deal = await api.getDeal(_id, userToken);
+
+    if (!deal || 'status' in deal) {
+      return res.redirect('/not-found');
+    }
+
+    if (!canSubmissionTypeBeCancelled(deal.dealSnapshot.submissionType)) {
```

Comment:
I'm wondering if the `canSubmissionTypeBeCancelled` check should be in middleware, we have this repeated in 4x places. What do you think?

Not a blocker for this PR.

Reply:
Its worth thinking about, but it needs the request to be made to get the deal so probably has to happen in the controller. Something I've been thinking about though - we could do something like the middleware calls the api, checks its valid, & passes this in as `req.body.deal` or similar. 
Would be a large change of pattern though!